tuna/miopen/subcmd/load_job.py

```diff
@@ -246,38 +190,46 @@ def add_jobs(args, dbt):
           counts += 1
         except IntegrityError as err:
           session.rollback()
-          LOGGER.warning('Integrity Error: %s', err)
+          logger.warning('Integrity Error: %s', err)
       if not do_commit:
         try:
           session.commit()
         except IntegrityError as err:
           session.rollback()
           counts = 0
           do_commit = True
-          LOGGER.warning(
+          logger.warning(
               'Quick update failed, rolling back to add one by one : %s', err)
           continue
       break
 
   return counts
 
 
-def main():
-  """ main """
-  args = parse_args()
+def run_load_job(args: argparse.Namespace, logger: logging.Logger):
+  """trigger the script run in miopen lib"""
+
   connect_db()
 
   dbt = MIOpenDBTables(session_id=None, config_type=args.config_type)
   if args.tag:
     try:
       test_tag_name(args.tag, dbt)
     except ValueError as terr:
-      LOGGER.error(terr)
-
-  cnt = add_jobs(args, dbt)
+      logger.error(terr)
+  if args.solvers or args.algo:
+    args = arg_solvers(args, logger)
 
+  cnt = add_jobs(args, dbt, logger)
   print(f"New jobs added: {cnt}")
```

Comment:
Make this also return the number of jobs it added, then you can use that to check the expected number in the utest